File: code/classes/extensions.ex
# Extensions in Elixir (using modules with functions)
defmodule StringExt do
  def reverse(str) do
    str
    |&gt; String.graphemes()
    |&gt; Enum.reverse()
    |&gt; Enum.join()
  end
end

original = "hello"
reversed = StringExt.reverse(original)
IO.puts reversed

File: code/classes/extensions.java
// Extensions in Java (using inheritance or utility classes)
class StringExtensions {
    public static String reverse(String str) {
        return new StringBuilder(str).reverse().toString();
    }
}

public class Main {
    public static void main(String[] args) {
        String original = "hello";
        String reversed = StringExtensions.reverse(original);
        System.out.println(reversed);
    }
}